R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
+            var.annot <- contrastDF %>% dplyr::select(Group, LogInt)
+            sizeBy <- "LogInt"
+            sizes <- c(4, 8, 10, 11, 12)
+            showSizeLegend <- TRUE
+        } else if (!missing(geneSymCol) && !sizeByIntensity) {
+            var.annot <- contrastDF %>% dplyr::select(c(Group,geneSymCol))
+            colnames(var.annot)[colnames(var.annot) %in% geneSymCol] <- "GeneName"
+            sizeBy <- "Group"
+            sizes <- sizes
+            showSizeLegend <- FALSE
+        } else if (!missing(geneSymCol) && sizeByIntensity) {
+            var.annot <- contrastDF %>% dplyr::select(Group, geneSymCol, LogInt)
+            colnames(var.annot)[colnames(var.annot) %in% geneSymCol] <- "GeneName"
+            sizeBy <- "LogInt"
+            sizes <- c(4, 8, 10, 11, 12)
+            showSizeLegend <- TRUE
+        } else {
+            var.annot <- contrastDF %>% dplyr::select(Group)
+            sizeBy  <- "Group"
+            sizes <- sizes
+            showSizeLegend <- FALSE
         }
 
-        foldChangeMargin <- (foldChangeLines + (foldChangeLines * 0.2))
-        volcanoPlot <- canvasXpress::canvasXpress(data                    = cx.data,
-                                                  varAnnot                = var.annot,
-                                                  decorations             = decorations,
-                                                  graphType               = "Scatter2D",
-                                                  colorBy                 = "Group",
-                                                  colors                  = symbolFill,
-                                                  legendPosition          = legendPosition,
-                                                  showDecorations         = TRUE,
-                                                  sizes                   = c(4, 10, 12, 14, 16, 18, 20, 22, 24, 26),
-                                                  sizeByShowLegend        = sizeByShowLegend,
-                                                  title                   = title,
-                                                  xAxisTitle              = xlab,
-                                                  yAxisTitle              = ylab,
-                                                  sizeBy                  = sizeBy,
-                                                  setMaxX                 = foldChangeMargin,
-                                                  setMinX                 = -1*foldChangeMargin,
-                                                  citation                = footnote,
-                                                  citationFontSize        = footnoteSize,
-                                                  citationColor           = footnoteColor,
-                                                  events                  = events)
+        inside_legend_only_positions <- c("topRight", "bottomRight", "topLeft", "bottomLeft")
```

Comment:
what does this mean ?

Reply:
These 4 positions are displaying legends inside the plots, So for CX plots there is one parameter which is insideLeged=TRUE.
So when any one of these 4 positions is called there must be call insideLeged=TRUE otherwise  insideLeged=FALSE.

insideLeged=FALSE for these positions (top,bottom,left,right).

Reply:
![image](https://user-images.githubusercontent.com/83666355/124708633-281b0280-df18-11eb-827b-5d7aad1150b8.png)